daemon/algod/api/server/v2/utils.go

```diff
@@ -431,12 +431,11 @@ func convertProgramTrace(programTrace []simulation.OpcodeTraceUnit) *[]model.Sim
 	return &modelProgramTrace
 }
 
-func digestOrNil(digest crypto.Digest) *string {
+func digestOrNil(digest crypto.Digest) *[]byte {
 	if digest.IsZero() {
 		return nil
 	}
-	digestStringFromBytes := string(digest.ToSlice())
-	return &digestStringFromBytes
+	return sliceOrNil(digest.ToSlice())
```

Comment:
nit: `sliceOrNil` not needed since `digest.ToSlice` should never produce a 0-length value

Reply:
but digest might be the zero digest?

Reply:
zero digest is omit to nil ptr in `digestOrNil`